homeassistant/components/evohome/climate.py

```diff
@@ -309,28 +314,77 @@ def max_temp(self) -> float:
         Controllers do not have a max target temp, but one is required by HA.
         """
         temps = [z.setpointCapabilities['maxHeatSetpoint']
-                 for z in self._evo_device._zones]  # noqa: E501; pylint: disable=protected-access
+                 for z in self._evo_device.zones.values()]
         return max(temps) if temps else 35
 
-    def _set_operation_mode(self, op_mode: str) -> None:
-        """Set the Controller to any of its native EVO_* operating modes."""
-        try:
-            self._evo_device._set_status(op_mode)  # noqa: E501; pylint: disable=protected-access
-        except (requests.exceptions.RequestException,
-                evohomeclient2.AuthenticationError) as err:
-            _handle_exception(err)
-
     def set_hvac_mode(self, hvac_mode: str) -> None:
         """Set an operating mode for the Controller."""
-        self._set_operation_mode(HA_HVAC_TO_TCS.get(hvac_mode))
+        self._set_tcs_mode(HA_HVAC_TO_TCS.get(hvac_mode))
 
     def set_preset_mode(self, preset_mode: Optional[str]) -> None:
         """Set a new preset mode.
 
         If preset_mode is None, then revert to 'Auto' mode.
         """
-        self._set_operation_mode(HA_PRESET_TO_TCS.get(preset_mode, EVO_AUTO))
+        self._set_tcs_mode(HA_PRESET_TO_TCS.get(preset_mode, EVO_AUTO))
 
     def update(self) -> None:
         """Get the latest state data."""
         pass
+
+
+class EvoThermostat(EvoZone):
+    """Base for a Honeywell Round Thermostat.
+
+    Implemented as a combined Controller/Zone.
+    """
+
+    def __init__(self, evo_broker, evo_device) -> None:
+        """Initialize the Round Thermostat."""
+        super().__init__(evo_broker, evo_device)
+
+        self._name = evo_broker.tcs.location.name
+        self._icon = 'mdi:radiator'
+
+        self._preset_modes = [PRESET_AWAY, PRESET_ECO]
+
+    @property
+    def device_state_attributes(self) -> Dict[str, Any]:
+        """Return the device-specific state attributes."""
+        status = super().device_state_attributes['status']
+
+        status['systemModeStatus'] = getattr(self._evo_tcs, 'systemModeStatus')
+        status['activeFaults'] += getattr(self._evo_tcs, 'activeFaults')
+
+        return {'status': status}
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return the current operating mode."""
+        if self._evo_tcs.systemModeStatus['mode'] == EVO_HEATOFF:
+            return HVAC_MODE_OFF
+
+        return super().hvac_mode
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return the current preset mode, e.g., home, away, temp."""
+        if self._evo_tcs.systemModeStatus['mode'] == EVO_AUTOECO:
+            if self._evo_device.setpointStatus['setpointMode'] == EVO_FOLLOW:
```

Comment:
Combine these two checks with `and`.

Reply:
PR merged - will add it to the next PR (there is sure to be one)